components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/impl/FaultyRequestDataPublisher.java

```diff
@@ -26,13 +29,19 @@
  */
 public class FaultyRequestDataPublisher extends AbstractRequestDataPublisher {
 
+    private static final Log log = LogFactory.getLog(FaultyRequestDataPublisher.class);
     @Override
     public CounterMetric getCounterMetric() {
-        return AnalyticsDataPublisher.getInstance().getFaultyMetricReporter();
+        return null;
```

Comment:
we are not gonna use this method? have we checkd for all the places where we use this

Reply:
It is only used in one place within the carbon distribution..
Haven't checked if it is called from outside
What do you suggest Tishan ?

Reply:
in the location where we call this did we change that to use the new method?

Reply:
Yes Tishan.
Here it is 
https://github.com/SarithRavI/carbon-apimgt/blob/parallel-sep/components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/impl/AbstractRequestDataPublisher.java#L49